### ISSUE TITLE
Turn off notifications for hearts

### ISSUE THREAD

**Issue by emgoto** [forem/forem#3244]
**Is your feature request related to a problem? Please describe.**
I would like to turn off being notified (by the number icon in the top right corner) when I have received a heart/like on a comment or article.

**Describe the solution you'd like**

When you go to the notification settings page, there's a checkbox option to turn off receiving notifications for hearts / likes. Ideally I'd like to be notified when someone comments / replies to a comment but nothing else.

**Comment by Zhao-Andy** [comment]
Thanks for the feature request! We'll discuss this as a team and leave another comment.

**Comment by maestromac** [comment]
Very similar to https://github.com/thepracticaldev/dev.to/issues/3171 .

**Comment by nicolas-amabile** [comment]
Hello everyone!
Here is my first try to implement this feature: https://github.com/thepracticaldev/dev.to/pull/3391
@rhymes' comment makes sense. I would like to discuss the approach and to get some help with some missing points on my PR.
- Does the same feature apply for organizations? I think it makes sense. I added the `fetch` and default value [here](https://github.com/thepracticaldev/dev.to/pull/3391/files#diff-029bf5e69e07f2f3c4eef3c94769633fR21) to handle the missing field, but I guess I need to add it to the organization as well.
- I would like to get some help with:
   - Migration script for adding `post_reaction_notifications` field
   - Tests
   - UI. Check the gif on the PR (#3391) description for the current implementation

**Comment by rhymes** [comment]
@nicolas-amabile 

> Does the same feature apply for organizations? I think it makes sense. I added the fetch and default value here to handle the missing field, but I guess I need to add it to the organization as well.

It does, if you look at [ReactionsController#create](https://github.com/thepracticaldev/dev.to/blob/master/app/controllers/reactions_controller.rb#L41) method, you'll see that both the user and the organization can receive a reaction notification:

https://github.com/thepracticaldev/dev.to/blob/727d2fadd049cca6f0cda8168488d40b9c615793/app/controllers/reactions_controller.rb#L65-L66

The [fetch](https://github.com/thepracticaldev/dev.to/pull/3391/files#diff-029bf5e69e07f2f3c4eef3c94769633fR21) you added in that case wouldn't work because as we said in the PR, the field in not in the `receiver_data` hash, but on the `receiver` model

> Migration script for adding post_reaction_notifications field

You can take a look at the guide on how to create a migration for Rails: https://guides.rubyonrails.org/active_record_migrations.html#creating-a-standalone-migration

You modified the `schema.rb` manually but you don't need to do that, it will be generated automatically when, after you wrote the migration script, you [apply it locally](https://guides.rubyonrails.org/active_record_migrations.html#running-migrations) with `rails db:migrate`

> Tests

You probably want to add tests for two things:

* the async job, to test that only if the flag is true the job is ran, you can find its tests here https://github.com/thepracticaldev/dev.to/blob/master/spec/jobs/notifications/new_reaction_job_spec.rb

* that the user can active or deactivate the flag in the settings page, you can see its tests here https://github.com/thepracticaldev/dev.to/blob/master/spec/requests/user_settings_spec.rb#L42

BTW I'd call the flag `article_SOMETHING`, mostly because `Article` is how the post is called inside the code.

Hope this helps!

**Comment by emgoto** [comment]
👋 I'd like to give this one a crack

**Comment by rhymes** [comment]
Go ahead @emgoto, thank you!

**Comment by emgoto** [comment]
How do I get the spec to run? If I run `bundle exec rspec spec/workers/notifications/new_reaction_worker_spec.rb` it times out:

```
An error occurred in a `before(:suite)` hook.
Failure/Error: next unless Search::Client.indices.exists(index: search_class::INDEX_NAME)

Faraday::TimeoutError:
  Operation timed out after 30004 milliseconds with 0 bytes received
# ./app/services/search/cluster.rb:49:in `block in delete_indexes'
# ./app/services/search/cluster.rb:48:in `each'
# ./app/services/search/cluster.rb:48:in `delete_indexes'
# ./app/services/search/cluster.rb:14:in `recreate_indexes'
# ./spec/rails_helper.rb:91:in `block (2 levels) in <top (required)>'
# ------------------
# --- Caused by: ---
# Patron::TimeoutError:
#   Operation timed out after 30004 milliseconds with 0 bytes received
#   ./app/services/search/cluster.rb:49:in `block in delete_indexes'
```

Nicolas added the `post_reaction_notifications` to the "users" table but also mentioned something about organisations, do I need to add to another table as well?

**Comment by rhymes** [comment]
@emgoto can you make sure you have started Elasticsearch? What happens if you type `
curl http://localhost:9200` on the command line? If it doesn't work you should make sure the ES server is up:

- `brew services start elasticsearch-oss` if you have macOS
- https://www.elastic.co/guide/en/elasticsearch/reference/7.5/targz.html#install-linux for Linux

**Comment by emgoto** [comment]
You're right, I didn't have that running. Thanks!

My other question - Nicolas added the `post_reaction_notifications` to the "users" table but also mentioned something about organisations, I'm wondering how the two relate to each other.

Do I also need to add `post_reaction_notifications` to the `Organizations` table? In that case how would I update it from the UI?

Alternatively do I always send out the notification for organizations, and only do the check for users?

**Comment by cmgorton** [comment]
It looks like this issue was solved with #9435. I am going to go a head and close this issue.
